### PR TITLE
Microsoft.Migrate - Split the existing readme(s) into separate files based on service

### PR DESCRIPTION
**Pull request by srilipta-swain**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - For smooth onboarding of customers for respective service, we intend to take this change up where we split the existing readme files to respective service's readme. This is going to make the SDK generation process smooth.

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

## Namespace approval links:

- [OffAzure](https://github.com/Azure/azure-sdk/issues/7863)
- [Migrate](https://github.com/Azure/azure-sdk/issues/6965)

### CODE REVIEW

**Comment by ajaygit158** [comment]
My understanding is that Migrate wants to have separate SDKs for each of the sub services.
Can we keep the original readme files same and avoid modifications to the same in future considering that we dont want to release Migrate as a combined SDK?
You can add new project/RT specific readme files for the new API versions and reference your RT specific SDKs generation to those RT specific readme files?

**Comment by srilipta-swain** [comment]
> My understanding is that Migrate wants to have separate SDKs for each of the sub services. Can we keep the original readme files same and avoid modifications to the same in future considering that we dont want to release Migrate as a combined SDK? You can add new project/RT specific readme files for the new API versions and reference your RT specific SDKs generation to those RT specific readme files?

Discussed offline and adding the old readmes as is. Will not add new versions to older readmes.

**Comment by ajaygit158** [comment]
As discussed offline, please review the namespace for the packages. Remove the ARMChangesRequested tag post that.

File: specification/migrate/resource-manager/Microsoft.OffAzure/readme.go.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+module: github.com/Azure/azure-sdk-for-go/$(module-name)
+output-folder: $(go-sdk-folder)/$(module-name)
+azure-arm: true
+```
+
+### Go multi-api
+
+```yaml $(go) && $(multiapi)
+batch:
+  - tag: package-2020-01
```
Comment (jhendrixMSFT) [inline specification/migrate/resource-manager/Microsoft.OffAzure/readme.go.md:24]: This tag doesn't match any existing tags in the `readme.md` file which is why no input files can be found.

The value must align with the tag you intend to build.

Comment applies to all `readme.go.md` files.

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.OffAzure/readme.go.md:24]: Just curious, why did others threw no error ?

Comment (jhendrixMSFT) [reply specification/migrate/resource-manager/Microsoft.OffAzure/readme.go.md:24]: Looks like they did. See [python build log](https://github.com/Azure/azure-rest-api-specs/pull/30705/checks?check_run_id=32276795935).

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.OffAzure/readme.go.md:24]: @jhendrixMSFT Thanks for the quick help. Even after updating the packages name I'm still getting the same error.

**Comment by jhendrixMSFT** [comment]
@lirenhe @tadelesh can one of you please take a look? Do we have a template for the readme config files?

File: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+> see https://aka.ms/autorest
+
+This is the AutoRest configuration file for Azure Migrate.
+
+---
+
+### Java multi-api
+
+``` yaml $(java) && $(multiapi)
```
Comment (msyyc) [inline specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md:11]: Please put java configuration into readme.java.md like other languages.
Add JAVA SDK owner @XiaofeiCao for awareness.

Comment (XiaofeiCao) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md:11]: Thanks Yuchao.

Java doesn't use multiapi anymore, though we still want to have a separate readme.java.md. This seems first beta for Java SDK and we may want to disable model flattening in that file.

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md:11]: @XiaofeiCao thanks! Can you please share any reference to similar file which has disabled model flattening? or any documentation that can give me an idea? Also, curious what does disabling model flattening helps us with ?

Comment (XiaofeiCao) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md:11]: > @XiaofeiCao thanks! Can you please share any reference to similar file which has disabled model flattening? or any documentation that can give me an idea? Also, curious what does disabling model flattening helps us with ?

Ah, I mean if you add readme.md.java, I could help add required settings for Java SDK there. 
NVM, let me extract java related settings, as the old template is out-of-date.

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md:11]: Thanks @XiaofeiCao , please do share. Meanwhile let me commit the readme.java.md files.

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: Just curious, why did other SDKs for Go threw no error?

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: Hi @XiaofeiCao I have added the JAVA readme files. But Go SDK build still fails. Can you please check ?

File: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.nodejs.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+## Node.js
+
```
Comment (lirenhe) [inline specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.nodejs.md:2]: I don't think we are using nodejs.md file now, could you remove all nodejs.md files in this PR?

File: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md (hunk 2 above)
Comment (XiaofeiCao) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: Thanks @srilipta-swain , appreciate it!

Looks like Go can't find input file, though the tag is there.. @tadelesh Could you help take a look?
I'll adjust Java settings and see we have the same issue.

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: Got it, basically the default tag version's input & description was missing.

File: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.nodejs.md (hunk 3 above)
Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.nodejs.md:2]: Sure, will remove - thanks.

File: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md (hunk 2 above)
Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: @XiaofeiCao seems like your comment didn't show up earlier. I have fixed the Go SDK build, thanks!

**Comment by tadelesh** [comment]
> @lirenhe @tadelesh can one of you please take a look? Do we have a template for the readme config files?

sample file is here: https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/samplefiles/readme.go.md. go do not need to set multi-api package config anymore. @srilipta-swain you could remove that part according to the template.

File: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md (hunk 2 above)
Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: @XiaofeiCao I have addressed all the concerns, can you please confirm if I can proceed with the PR merge request ?

**Comment by XiaofeiCao** [comment]
@raych1 Do you know why SDK automation is skipping Java SDK?

File: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md (hunk 2 above)
Comment (XiaofeiCao) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: Yes, please. Thanks!

Comment (srilipta-swain) [reply specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/readme.md]: Thanks for confirmation, will proceed.

**Review by v-anishrao — APPROVED** [review]

**Pull request merged**